pages/_app.tsx

```diff
@@ -1,20 +1,31 @@
-import React, { useState, FC } from 'react';
+import React from 'react';
 import { GeistProvider, CssBaseline } from '@geist-ui/react';
 import { wrapper } from '../redux/store';
 import { AppProps } from 'next/app';
 import Nav from '../components/Nav';
+import { useSelector } from 'react-redux';
+import { parseCookies } from 'nookies';
+import { setTheme, setApiKey } from '../redux/actions/settings.actions';
+import { getTheme } from '../redux/selectors/settings.selectors';
 
-import { setCookie, parseCookies } from 'nookies';
-
-const OpticTracker: FC<AppProps> = ({ Component, pageProps }): JSX.Element => {
-  const [useLight, setUseLight] = useState<boolean>(false);
+const OpticTracker = ({ Component, pageProps }: AppProps): JSX.Element => {
+  const theme = useSelector(getTheme);
 
   return (
-    <GeistProvider theme={{ type: useLight ? 'light' : 'dark' }}>
+    <GeistProvider theme={{ type: theme }}>
       <CssBaseline />
-      <Nav useLight={useLight} setUseLight={setUseLight} />
+      <Nav />
       <Component {...pageProps} toggleLight />
     </GeistProvider>
   );
 };
+
+OpticTracker.getInitialProps = async ({ ctx }) => {
```

Comment:
When we first load any page, we should check if cookies exist and if they do parse them and add them to the state.